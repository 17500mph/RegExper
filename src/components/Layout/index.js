import React from 'react';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';

import Header from 'components/Header';

const Layout = ({ title, children }) => <React.Fragment>
  <Helmet>
    <title>{ title ? `Regexper - ${ title }` : 'Regexper' }</title>
  </Helmet>
  <Header />
  { children }
</React.Fragment>;

Layout.propTypes = {
  title: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};

export default Layout;
