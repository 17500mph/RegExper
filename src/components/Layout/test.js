import React from 'react';
import { shallow } from 'enzyme';

import Layout from 'components/Layout';

describe('Layout', () => {
  test('rendering', () => {
    const component = shallow(
      <Layout>
        Example content
      </Layout>
    );
    expect(component).toMatchSnapshot();
  });

  test('rendering with a title', () => {
    const component = shallow(
      <Layout title="Testing">
        Example content
      </Layout>
    );
    expect(component).toMatchSnapshot();
  });
});
