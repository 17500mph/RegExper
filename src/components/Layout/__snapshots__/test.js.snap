// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout rendering 1`] = `
<SentryBoundary>
  <noscript>
    <style
      type="text/css"
    >
      
  [data-requires-js] {
    display: none !important;
  }

    </style>
  </noscript>
  <withI18nextTranslation(Header)
    banner="Test Banner"
  />
  <SentryBoundary>
    Example content
  </SentryBoundary>
  <withI18nextTranslation(Footer)
    buildId="test-buildid"
  />
</SentryBoundary>
`;
